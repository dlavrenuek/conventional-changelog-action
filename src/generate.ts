import path from "node:path";
import { type Changelog, generateChangelog } from "./changelog";
import { getConfig } from "./config";
import { getCommits, getIssuesPath } from "./git";

type Generate = (params: {
  from: string;
  to: string;
  configFile: string;
}) => Promise<Changelog>;

const generate: Generate = async ({ from, to, configFile }) => {
  const config = getConfig(
    configFile || path.join(__dirname, "defaultConfig.json"),
  );
  const issuesUrl = config.issuesUrl || (await getIssuesPath());
  const commits = await getCommits({ from, to });
  return generateChangelog({
    ...config,
    commits,
    issuesUrl,
  });
};

export default generate;
