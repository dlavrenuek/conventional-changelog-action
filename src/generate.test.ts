import generate from "./generate";

// @ts-ignore
global.AbortSignal = () => null;

describe("generate.ts", () => {
  it("generates the correct changelog", async () => {
    const changelog = await generate({
      from: "3a145b38feb6c24e9d54b507aa121676b1f184ea",
      to: "34ead995676dcc41c5cf8d9400af67c432a58333",
      configFile: "",
    });

    expect(changelog).toMatchSnapshot();
  });
});
