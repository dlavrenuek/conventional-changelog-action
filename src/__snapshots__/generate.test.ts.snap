// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`generate.ts generates the correct changelog 1`] = `
{
  "body": "## ðŸš€ New Features
* add issuesUrl to configuration

## ðŸ’Š Bugfixes
* fix eslint rules and formatting

## ðŸ§¹ Chore
* fix formatting with prettier
* update usage documentation",
  "bump": "minor",
}
`;
